lib/peerconnection.js

```diff
@@ -199,75 +199,169 @@ function RTCPeerConnection(configuration, constraints) {
     }
   });
 
-  //
-  // PeerConnection methods
-  //
-
-  this.createOffer = function createOffer(successCallback, failureCallback, options) {
-    options = options || {};
+  this.createOffer = function createOffer(){
+    var call = function call(options, successCallback, failureCallback) {
+      queueOrRun({
+        func: 'createOffer',
+        args: [options],
+        wait: true,
+        onSuccess: function(sdp) {
+          successCallback.call(this, new RTCSessionDescription({type: 'offer', sdp: sdp}));
+        },
+        onError: failureCallback
+      });
+    };
 
-    // FIXME: complain if onError is undefined
-    queueOrRun({
-      func: 'createOffer',
-      args: [options],
-      wait: true,
-      onSuccess: function(sdp) {
-        successCallback.call(this, new RTCSessionDescription({type: 'offer', sdp: sdp}));
-      },
-      onError: failureCallback
-    });
+    if (arguments.length <= 1 && (typeof arguments[0]) === 'object') {      
+      // Promise-based call.
+      var options = (arguments.length === 1) ? arguments[0] : {};
+      return new Promise(function createOfferPromise(resolve, reject) {
+        call(options, resolve, reject);
+      });
+    } else if (arguments.length >= 2 && (typeof arguments[0]) === 'function' && (typeof arguments[1]) === 'function') {
+      // Legacy method.
+      var options = (arguments.length === 3) ? arguments[2] : {};
+      call(options, arguments[0], arguments[1]);
+    } else {
+      throw new Error('Invalid call to createOffer - function must either have prototype'
+        + ' ([config]) or (successCallback, failureCallback, [config]).');
+    }
   };
 
-  this.createAnswer = function createAnswer(successCallback, failureCallback, options) {
-    options = options || {};
+  this.createAnswer = function createAnswer(){
+    var call = function call(successCallback, failureCallback) {
+      queueOrRun({
+        func: 'createAnswer',
+        args: [{}],
+        wait: true,
+        onSuccess: function(sdp) {
+          successCallback.call(this, new RTCSessionDescription({type: 'answer', sdp: sdp}));
+        },
+        onError: failureCallback
+      });
+    };
 
-    // FIXME: complain if onError is undefined
-    queueOrRun({
-      func: 'createAnswer',
-      args: [options],
-      wait: true,
-      onSuccess: function(sdp) {
-        successCallback.call(this, new RTCSessionDescription({type: 'answer', sdp: sdp}));
-      },
-      onError: failureCallback
-    });
+    if (arguments.length === 0) {      
```

Comment:
This does not allow calling `createAnswer` with the optional RTCAnswerOptions. I'm getting

```
> pc.createAnswer({})
Error: Invalid call to createAnswer - function must either have prototype (void) or (successCallback, failureCallback).
    at RTCPeerConnection.createAnswer (/Users/mroberts/src/personal/node-webrtc/lib/peerconnection.js:253:13)
    at repl:1:4
    at REPLServer.defaultEval (repl.js:252:27)
    at bound (domain.js:287:14)
    at REPLServer.runBound [as eval] (domain.js:300:12)
    at REPLServer.<anonymous> (repl.js:417:12)
    at emitOne (events.js:82:20)
    at REPLServer.emit (events.js:169:7)
    at REPLServer.Interface._onLine (readline.js:210:10)
    at REPLServer.Interface._line (readline.js:549:8)
```

I think it could be rewritten to something like

```js
if (arguments.length === 0 || arguments.length === 1 && typeof arguments[0] === 'object') {
```